sub/sd_ass.c

```diff
@@ -281,21 +279,20 @@ static void decode(struct sd *sd, struct demux_packet *packet)
     } else {
         // Note that for this packet format, libass has an internal mechanism
         // for discarding duplicate (already seen) packets.
+        char *ass_line = packet->buffer;
+        int ass_len = packet->len;
         if (sd->opts->sub_filter_SDH) {
-            char *indata = talloc_strndup(NULL, packet->buffer, packet->len);
-            char *ass_line = filter_SDH(sd, track->event_format, 1, indata);
+            char *indata = talloc_strndup(NULL, ass_line, ass_len);
```

Comment:
I assumed that in this case the data in packet->buffer is not \0 terminated.
And as the filter expects a \0 terminated string I have to duplicate the data and \0 terminate it.